analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("% Mito /Cell") + 
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+```
+So this is really informative. It looks like SCPCR000003 is a very poor sample across all tools and has a lot of dead cells. It might be a good idea to remove that from our comparisons and not consider this for benchmarking. Additionally, it looks like we can see where our outlier in Alevin is, SCPCR000006, has a much higher mito content in the Alevin run than in other tools. Let's see if other metrics are affected with that sample. 
+
+```{r}
+# remove SCPCR000003 from cell_qc for the rest of the plotting 
+cell_qc_filter <- cell_qc %>%
+  dplyr::filter(sample != "SCPCR000003")
+```
+
+Now let's look at the number of UMI's detected/cell that we've removed our outlier sample. 
+
+```{r}
+cell_qc_filter %>%
+  mutate(tool = fct_reorder(tool, sum, .fun = 'mean')) %>%
+  ggplot(aes(x = reorder(tool, sum), y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  theme_classic() + 
+  ylab("Number of UMI/Cell") + 
+  xlab("")
+```
+From this, it looks like Cellranger has the lowest UMI's/cell with Alevin and kallisto very close to each other at the top. Alevin does have quite a large variation though. 
+
+Out of curiosity, how would it look if we kept that sample in that had really high mito content. 
+
+```{r}
+cell_qc %>%
+  mutate(tool = fct_reorder(tool, sum, .fun = 'mean')) %>%
+  ggplot(aes(x = reorder(tool, sum), y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  theme_classic() + 
+  ylab("Number of UMI/Cell") + 
+  xlab("")
+```
+You can see that the one outlier has very low UMI's/cell which brings alevin below all of the other methods, when in fact it looks like it might have the highest UMI's/cell for samples that are not all dead. In general, Alevin appears to have a lot more variability. Let's look at each sample individually. 
+
+```{r}
+# number of UMI's/sample
+ggplot(cell_qc_filter, aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("")
+```
+So, it looks like across all samples, cellranger is consistently near the bottom, while kallisto is consistently near the top. Alevin fluctuates all the way from detecting the most to detecting the least... 
+Alevin-fry seems to always be in the middle and it looks like the `--sketch` mode leads to increased UMI/cell with the same number of cells (see previous plots looking at the number of cells).
+
+What about genes/cell? Are the same trends consistent?
+
+```{r}
+cell_qc_filter %>%
+  mutate(tool = fct_reorder(tool, detected, .fun = 'mean')) %>%
+  ggplot(aes(x = reorder(tool, detected), y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  theme_classic() + 
+  ylab("Number of Genes detected/Cell") + 
+  xlab("")
+```
+
+
+```{r}
+ggplot(cell_qc_filter, aes(y = detected, fill = tool)) + 
```

Comment:
Again here, I think the plots split by sample are probably all that we need. Aggregating across the samples hides info that I think we want to see, and having both plots seems redundant. If there are no obvious trends from the separated samples, I would probably argue we can't make a good decision about the merits of different tools.